File: swift/Sources/Encode-TestDesign-Kata/ByteBuffer.swift
class ByteBuffer {
    private var buffer: [Int] = []
    private var current = 0

    func append(_ i: Int) {
        buffer.append(i)
    }

    func read() -> Int? {
        guard current < buffer.count else {
            return nil
        }
        let value = buffer[current]
        current += 1
        return value
    }

    func getAvailable() -> Int {
        return buffer.count - current
    }
}

File: js-jasmine/src/ByteBuffer.js
class ByteBuffer {
    constructor() {
        this.buffer = [];
        this.current = 0;
    }

    append(i) {
        this.buffer.push(i);
    }

    read() {
        if (this.current < this.buffer.length) {
            const value = this.buffer[this.current];
            this.current += 1;
            return value;
        } else {
            return undefined;
        }
    }

    getAvailable() {
        return this.buffer.length - this.current;
    }
}

module.exports = { ByteBuffer };
